chaco/base_contour_plot.py

```diff
@@ -54,10 +54,10 @@ class BaseContourPlot(Base2DPlot):
     # ------------------------------------------------------------------------
 
     # Is the cached level data valid?
-    _level_cache_valid = Bool(False)
+    _level_cache_valid = Bool(False, transient=True)
 
     # Is the cached color data valid?
-    _colors_cache_valid = Bool(False)
+    _colors_cache_valid = Bool(False, transient=True)
```

Comment:
If the cache valid flags arent transient, then the caches themselves should also be transient - `_levels` and `_colors`.

I would think `HasTraits` would ignore traits that are private i.e. that start an `_`. If that isn't the case, there are a lot more private traits that we will want to mark as transient.

Reply:
As you suggested, I printed `state` inside the `ArrayDataSource.__getstate__` method on master and ran the test suite.  I do see `_` prefixed traits, eg `_data`, '_min_index', etc.  So it looks like these need to be addressed

Reply:
>If that isn't the case, there are a lot more private traits that we will want to mark as transient.

Does this mean all private traits be transient?

Reply:
> As you suggested, I printed state inside the ArrayDataSource.__getstate__ method on master and ran the test suite. I do see _ prefixed traits, eg _data, '_min_index', etc. So it looks like these need to be addressed

Sigh. I was afraid of that.

> Does this mean all private traits be transient?

Let's not do that. Definitely not in this PR.

Reply:
So, I imagine the code will recreate the cache if a `*cache*` is `False`, which it is by default because we're marking it as a transient. So, I think we can ignore all of the traits that are recalculated based on the `*cache*` flag. - i.e. nothing actionable in this PR.